File: src/main/java/com/jrmplugin/core/ProjectStoreComponent.java
package com.jrmplugin.core;

import com.intellij.ide.util.PropertiesComponent;

public class ProjectStoreComponent {

    private final PropertiesComponent store;

    public ProjectStoreComponent() {
        this.store = PropertiesComponent.getInstance();
    }

    public void add(String key, String value) {
        store.setValue(key, value);
    }

    public String get(String key) {
        return store.getValue(key);
    }

}

File: src/main/java/com/jrmplugin/core/ProjectStoreComponent.kt
package com.jrmplugin.core

import com.intellij.ide.util.PropertiesComponent

class ProjectStoreComponent {
    private val store: PropertiesComponent = PropertiesComponent.getInstance()
    fun add(key: String, value: String) {
        store.setValue(key, value)
    }

    operator fun get(key: String): String? {
        return store.getValue(key)
    }

}